File: python/two-pointer.py
def is_palindrome(word:str)->bool:
    word=word.lower()
    left_pointer=0
    right_pointer=len(word)-1
    while left_pointer <= right_pointer:
        if word[left_pointer] != word[right_pointer]:
            return False
        left_pointer+=1
        right_pointer-=1
    return True

File: rust/two pointer.rs
fn is_palindrome(word: &str) -> bool {
    let chars: Vec<char> = word.chars().collect();
    let mut left = 0;
    let mut right = chars.len().saturating_sub(1);
    while left < right {
        if chars[left] != chars[right] {
            return false;
        }
        left += 1;
        right -= 1;
    }
    true
}
